### PR TITLE
Change log timestamp to ISO8601 w/ timezone

### PR DESCRIPTION
**Pull request by mstemm**
This prevents ambiguity when, say, running falco in a container, which
may have a different /etc/localtime than the host system.

### CODE REVIEW

**Comment by daixiang0** [comment]
Fix #527

**Comment by mfdii** [comment]
@lorenzo-david can you review for @mstemm

**Comment by mstemm** [comment]
Actually, this still requires some work before it's ready to review.

File: userspace/falco/logger.cpp [hunk 1]
```diff
@@ -123,9 +129,18 @@ void falco_logger::log(int priority, const string msg)
 		}
 
 		std::time_t result = std::time(nullptr);
-		string tstr = std::asctime(std::localtime(&result));
-		tstr = tstr.substr(0, 24);// remove trailling newline
-		fprintf(stderr, "%s: %s", tstr.c_str(), copy.c_str());
+		if(falco_logger::time_format_iso_8601)
+		{
+			char buf[sizeof "YYYY-MM-DDTHH:MM:SS-0000"];
+			strftime(buf, sizeof(buf), "%FT%T%z", std::gmtime(&result));
```
Comment (lorenzo-david) [inline userspace/falco/logger.cpp]: Might be worth checking return value of strftime.

File: userspace/falco/logger.cpp [hunk 2]
```diff
@@ -123,9 +129,18 @@ void falco_logger::log(int priority, const string msg)
...
+			fprintf(stderr, "%s: %s", buf, msg.c_str());
+		}
+		else
+		{
+			string tstr = std::asctime(std::localtime(&result));
```
Comment (lorenzo-david) [inline userspace/falco/logger.cpp]: I guess std::asctime can return empty tstr in case of error.

File: userspace/falco/logger.cpp (hunk 1 above)
Comment (mstemm) [reply userspace/falco/logger.cpp]: ok will do.

File: userspace/falco/logger.cpp (hunk 2 above)
Comment (mstemm) [reply userspace/falco/logger.cpp]: yep will fix.

**Pull request merged**